pkg/mod/fs.go

```diff
@@ -0,0 +1,80 @@
+package mod
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/anz-bank/sysl/pkg/syslutil"
+	"github.com/spf13/afero"
+)
+
+type Fs struct {
+	fs afero.Fs
+}
+
+func NewFs(fs afero.Fs) *Fs {
+	return &Fs{fs: fs}
+}
+
+func (fs *Fs) Open(name string) (afero.File, error) {
+	f, err := fs.fs.Open(name)
+	if err == nil {
+		return f, nil
+	}
+
+	mod, err := Find(name)
+	if err != nil {
+		return nil, err
+	}
+	relpath, err := filepath.Rel(mod.Name, name)
+	if err != nil {
+		return nil, err
+	}
+
+	return syslutil.NewChrootFs(afero.NewOsFs(), mod.Dir).Open(relpath)
```

Comment:
This is a bit tricky. I'm not super comfortable with this because it opens up the entire filesystem for access, but yeah, the filesystem which the modules need to come from likely will be different from the chroot the module itself comes from.

I think the solution is to pass 2 afero.Fs into this filesystem, one which is used for the non-module files, the second is for the modules,  this also allows for easy testing (you can easily test that files were written to a MemFs in the right place)

Reply:
I agree with it. But the problem here is that the root of the fs is `mod.Dir`, which is mutable for every different module. It cannot be confirmed when `cmdRunner.Run`. Do you have any ideas?

Reply:
Doesn't `go get` download the files into a common root directory? so `import github.com/somewhere/foo.sysl` would end up in `<known location>/github.com/somewhere`? The fs for that could then be a chrootFS at <known location>

Reply:
`go get` downloads the files into `$GOPATH/pkg/mod/github.com/org/repo@version`. Golang hard codes `$GOPATH + "pkg/mod"`, but we don't know `repo@version` at first.